terraform-provider-constellation/internal/provider/attestation_data_source.go

```diff
@@ -224,83 +155,11 @@ func (d *AttestationDataSource) Read(ctx context.Context, req datasource.ReadReq
 			return
 		}
 	}
-	tfMeasurements := convertMeasurementsTfStateCompatible(fetchedMeasurements)
-	diags := resp.State.SetAttribute(ctx, path.Root("measurements"), tfMeasurements)
+	tfAttestation.Measurements = convertToTfMeasurements(fetchedMeasurements)
```

Comment:
```suggestion
	tfAttestation.Measurements = convertToTfMeasurements(fetchedMeasurements)

```
nit: makes the separation a little more apparent